src/ScheduleFunctions.cpp

```diff
@@ -2026,15 +2026,22 @@ void validate_fused_group_schedule_helper(const string &fn,
         for (int i = 0; i < n_fused; ++i) {
             const Dim &d1 = dims_1[start_fuse_1 + i];
             const Dim &d2 = dims_2[start_fuse_2 + i];
-            bool equal = var_name_match(d1.var, d2.var) &&
-                         (d1.for_type == d2.for_type) &&
-                         (d1.device_api == d2.device_api) &&
-                         (d1.dim_type == d2.dim_type);
-            if (!equal) {
-                user_error << "Invalid compute_with: dims " << i << " of " << p.func_1 << ".s"
-                           << p.stage_1 << "(" << dims_1[start_fuse_1 + i].var << ") and " << p.func_2
-                           << ".s" << p.stage_2 << "(" << dims_2[start_fuse_2 + i].var << ") do not match.\n";
-            }
+            user_assert(var_name_match(d1.var, d2.var)) << "Invalid compute_with: names of dim "
+                                                        << i << " of " << p.func_1 << ".s"
+                                                        << p.stage_1 << "(" << dims_1[start_fuse_1 + i].var << ") and " << p.func_2
```

Comment:
The repetition is unfortunate but I don't see a simple way to remove it. You don't want to eagerly produce the string in the non-error case.

But there's something weird in the existing code that looks like it could be shorter: dims_1[start_fuse_1 + i].var could be d1.var

Reply:
Good point, fixed.